lib/rubocop/cop/rails/migration_timestamp.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'time'
+
+module RuboCop
+  module Cop
+    module Rails
+      # Checks that migration file names start with a valid timestamp.
+      #
+      # @example
+      #   # bad
+      #   # db/migrate/bad.rb
+
+      #   # bad
+      #   # db/migrate/123_bad.rb
+
+      #   # bad
+      #   # db/migrate/20171301000000_bad.rb
+      #
+      #   # good
+      #   # db/migrate/20170101000000_good.rb
+      #
+      class MigrationTimestamp < Base
+        include RangeHelp
+
+        MSG = 'Migration file name must start with `YYYYmmddHHMMSS_` timestamp.'
+
+        def on_new_investigation
+          file_path = processed_source.file_path
+          return unless file_path.include?('db/migrate')
+
+          timestamp = File.basename(file_path)[/\A\d{14}(?=_)/]
+          return if valid_timestamp?(timestamp)
+
+          add_offense(source_range(processed_source.buffer, 1, 0))
+        end
+
+        private
+
+        def valid_timestamp?(timestamp, format: '%Y%m%d%H%M%S')
```

Comment:
What about checking that the timestamp is not in the future? I think the original discussion we had was triggered by migrations that had timestamps way in the future, which was causing problems.